### PR TITLE
 Remove the 'add new payment methods' item from the tasks list

### LINKED ISSUE #3060: Remove duplicative task list item for additional payment methods
the payment methods early access task list item is duplicative with the inbox note, let's remove the tasklist item in favor of the inbox note. 

Screen Shot 2021-10-06 at 7 54 18 PM

cc: @elizaan36

### PR DESCRIPTION
**Pull request by a-danae**
Fixes #3060 

This is the first iteration of this approach, planning to adjust the tests after getting some feedback 🙂 

#### Changes proposed in this Pull Request

* Remove the task for adding new payment methods from the overview task list. Both from WooCommerce's and WCPay's lists.
* Register a new page for setting up the new payment methods, with the same content the task page previously had.
* Redirect to the newly added page instead of the task one when enabling UPE via the inbox noticed and via the settings page.

As mentioned in the issue, this item is duplicative with the inbox note and should be removed in favor of the inbox note.

WCPay before and after:
image

WC before and after:
image

#### Testing instructions

**On WCPay:**

On the overview page:
* From the WCPay dev tools, disable UPE and enable the account overview task list
* Go to Payments -> Overview
* Under the "Things to do" list, the "Boost your sales by accepting new payment methods" item must not show up, but the inbox notice with the same title must still be there

On the overview page as well:
* On the inbox notice, clicking on "Boost your sales by accepting new payment methods" should take you to `admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods` and [that page](https://d.pr/i/JiS9PC) must not be blank

On the settings page:
* Go to Payments -> Settings
* Scroll to "Payments accepted on checkout" and disable it if it was enabled
* Enable it again by clicking on "Enable in your store"
* It should take you to `admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods` and [that page](https://d.pr/i/JiS9PC) must not be blank.

**On WC:**

On the home page:
* From the WCPay dev tools, disable UPE
* Go to WooCommerce -> Home
* Under the "Things to do next" list, the "Boost your sales by accepting new payment methods" item must not show up, but the inbox notice with the same title must still be there

On the home page as well:
* On the inbox notice, clicking on "Boost your sales by accepting new payment methods" should take you to `admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods` and [that page](https://d.pr/i/JiS9PC) must not be blank.

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

File: includes/admin/class-wc-payments-admin.php
Comment (a-danae) [inline includes/admin/class-wc-payments-admin.php:273]: There's probably a more convenient path for this screen. I've added this one in the meantime but glad to update it.

**Comment by htdat** [comment]
@a-danae - you're still waiting for reviews from @Automattic/gamma, right? 
It has not been added into our ZenHub board for the current sprint, so we were not aware of it. I've done that so that we can review it soon.

**Comment by a-danae** [comment]
@htdat, yess. I was planning to get someone from our team to review it first thinking you were busy. Thought that just adding Gamma as the reviewer would do, but please let me know if there's a different process to get a review from you. For next time 🙂 

Thanks for picking this up!

**Comment by a-danae** [comment]
Following what Gamma suggested, we're reviewing this within Axiom first, and we'll ping Heinsenberg next, as it looks like they authored the sections that were updated in this PR. Thanks again!

File: client/additional-methods-setup/index.js
Comment (malithsen) [inline client/additional-methods-setup/index.js:29]: Is there a situation where we show this page with UPE off, making `<MethodSelector />` appear?

**Review by malithsen — COMMENTED** [review]
Hi @a-danae this works as described and I didn't notice any issues testing this locally. 

I was almost going to request changes on the failing test cases before noticing your note on updating tests after initial feedback. 

Had a question in the comments on when the `<MethodSelector>` would show up.

File: client/additional-methods-setup/index.js
Comment (a-danae) [reply client/additional-methods-setup/index.js:29]: I didn't find any "natural" flow where that'd happen. The wp_option `_wcpay_feature_upe_settings_preview`is true by default and isn't set nor updated in our code, which is the option that defines the `isUpeSettingsPreviewEnabled` variable here.

The reason why I added this conditional is to keep the behavior from the component I removed, [here](https://github.com/Automattic/woocommerce-payments/pull/3134/files/9e9a328e10e465ad92df0ad2f20c1393a5f8d01b#diff-64da8dc3c200ed8ee47daef3203920aa6d1554e509a9eda184c959844b90991bL57). By the time I checked, even when you disabled UPE preview, this module wasn't saving due to an error. I'll check again but if this is the case, we might be good removing it. We could check with Heisenberg if so.

**Comment by a-danae** [comment]
Hey @malithsen! Thanks for reviewing this and bringing that question up 🙂 I'll adjust the tests for getting another review round. Hoping to check whether we can remove that conditional you mentioned then as well.

**Comment by a-danae** [comment]
@malithsen tests adjusted. This is ready to review.

@Automattic/heisenberg I've requested you to review it because it looks like you've [worked near](https://github.com/Automattic/woocommerce-payments/blame/c5224edd16b665bc3b2e024972078030e761b3ea/client/additional-methods-setup/task.js) the places this PR touches. To double-check we aren't missing anything.

**Review by malithsen — APPROVED** [review]
Changes since the last review look good to me @a-danae. Nice work!

**Review by harriswong — APPROVED** [review]
- In WC Overview and WCPay Settings, I can confirm the link now links to a `page=`. 
- I enabled the `_wcpay_feature_account_overview_task_list` feature and can confirm "Boost your sales..." task is not listed in the "Things to do" list.
- I took a look at the PR (https://github.com/Automattic/woocommerce-payments/pull/2358/files) where we added this task. I think we don't need this line anymore https://github.com/Automattic/woocommerce-payments/pull/2358/files#diff-3863b4f88d0515e080fadc56af3c5e38e942afb96e569cb000ae6a594e8bea4aR277. Maybe we can remove it if it is no longer needed.
- I was also wondering if we still need the whole `additionalMethodsSetup` around https://github.com/Automattic/woocommerce-payments/pull/2358/files#diff-dcc31bbf337cce3033cea17561d762d4e2511007a0f70c81bdf168a271727516R40. Although we moved the "task" to a "page", the functionalities around tracking "additional methods setup" is still considered a task, so I think it makes sense to keep them around.

Beside the comment above regarding  `additionalMethodsSetup.isTaskVisible`, LGTM 👍

File: client/additional-methods-setup/index.js
Comment (a-danae) [reply client/additional-methods-setup/index.js:27]: Adding some more context here. I don't see paths that would take to this page when UPE preview is disabled after this PR is merged. And when you go there directly, UPE methods are displayed as available. Maybe we better remove `<MethodSelector />` and prevent this screen from being displayed if UPE preview is disabled? 

Asking here because we're on it, but we could probably address it in a separate PR.

Comment (harriswong) [reply client/additional-methods-setup/index.js:27]: When we first developed UPE and introduce a new settings redesign page, we [introduced](https://github.com/Automattic/woocommerce-payments/issues/2154) this feature flag `_wcpay_feature_upe_settings_preview` so that we can toggle between old settings and the new settings. We also later allowed users to enroll into this "new settings preview" to get some early feedback.

UPE and the "new settings UI" should be the default now as you mentioned. The additional UPE methods can only be accessed via our new settings UI when UPE is enabled.

I think the `isUpeSettingsPreviewEnabled` handling here is probably not needed. We can tackle this in another PR though, I don't think it blocks this from merging.

**Comment by a-danae** [comment]
Cool, I've created the issue https://github.com/Automattic/woocommerce-payments/issues/3434 to address:
* The unused `additionalMethodsSetup.isTaskVisible` variable mentioned in Harris' [review ](https://github.com/Automattic/woocommerce-payments/pull/3134#pullrequestreview-816998152)
* Check whether we still need the whole `additionalMethodsSetup`, mentioned in Harris' review as well
* Remove `<MethodsSelector />`, mentioned in Malith's [review](https://github.com/Automattic/woocommerce-payments/pull/3134#pullrequestreview-797039812) and discussed with Harris there

Changelog updated and proceeding to merge.

Thank you both, @malithsen and @harriswong, for checking this out! 🙂

**Pull request merged**
